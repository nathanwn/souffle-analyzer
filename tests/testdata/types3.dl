.type Name <: symbol

.decl is_student(name: Name, id: number)
.decl is_teacher(name: Name, id: number)

identity("Alice", 1).
identity("Bob", 2).

.decl has_id(name: Name)

has_id(name) :- is_student(name, _).
has_id(name) :- is_teacher(name, _).
