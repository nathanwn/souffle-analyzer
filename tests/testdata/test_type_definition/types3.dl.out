types3.dl
 0 | .type Name <: symbol
 1 |
 2 | .decl is_student(name: Name, id: number)
 3 | .decl is_teacher(name: Name, id: number)
 4 |
 5 | identity("Alice", 1).
 6 | identity("Bob", 2).
 7 |
 8 | .decl has_id(name: Name)
 9 |
10 | has_id(name) :- is_student(name, _).
11 | has_id(name) :- is_teacher(name, _).

=====

-- Cursor range --
types3.dl
10 | has_id(name) :- is_student(name, _).
   |        ~~~~
-- Type Definition --
types3.dl
 0 | .type Name <: symbol
   |       ~~~~
_____

-- Cursor range --
types3.dl
10 | has_id(name) :- is_student(name, _).
   |                            ~~~~
-- Type Definition --
types3.dl
 0 | .type Name <: symbol
   |       ~~~~
_____

-- Cursor range --
types3.dl
11 | has_id(name) :- is_teacher(name, _).
   |        ~~~~
-- Type Definition --
types3.dl
 0 | .type Name <: symbol
   |       ~~~~
_____

-- Cursor range --
types3.dl
11 | has_id(name) :- is_teacher(name, _).
   |                            ~~~~
-- Type Definition --
types3.dl
 0 | .type Name <: symbol
   |       ~~~~
_____
